File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        
        current_node = head

        while current_node.next:
            new_head = current_node.next
            current_node.next = new_head.next
            new_head.next = head
            head = new_head

        return head    

File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null) {
            return head;
        }
        
        var currentNode = head;
        
        while (currentNode.next != null) {
            
            final var newHead = currentNode.next;
            currentNode.next = newHead.next;
            newHead.next = head;
            head = newHead;
        }
        
        return head;
    }
}